client/src/main/java/io/pravega/client/security/auth/DelegationTokenHandlingStrategy.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.security.auth;
+
+/**
+ * Strategy for handling delegation tokens of various kinds, such as null, empty, JWT, non-JWT strings, etc.
+ */
+public interface DelegationTokenHandlingStrategy {
+
+    String retrieveToken();
```

Comment:
Javadoc for **all** publicly visible methods, especially on interfaces.

Reply:
Sure, I've added the JavaDocs in the latest commit I have pushed.